### PR TITLE
Kernel: propagate errors in many places

### PR DESCRIPTION
**Pull request by thomasqueirozb**
This also removes `Vector::grow` & `Vector::ensure_capacity` from the kernel

### CODE REVIEW

File: Kernel/Locking/Mutex.cpp [hunk 1]
```diff
@@ -249,19 +251,19 @@ ErrorOr<void> Mutex::block(Thread& current_thread, Mode mode, SpinlockLocker<Spi
     return {};
 }
 
-void Mutex::unblock_waiters(Mode previous_mode)
+ErrorOr<void> Mutex::unblock_waiters(Mode previous_mode)
```
Comment (bgianfo) [inline Kernel/Locking/Mutex.cpp:254]: I understand it was a `FIXME` that drove this change, but in this case the `FIXME` pointed you in the wrong direction. I don't think this is the correct route to go down. Mutex locking needs to be designed in a way that can't fail. Having a `Mutex::unlock()` that can fail makes it impossible to reason about the state of the system.

**Review by bgianfo — CHANGES_REQUESTED** [review]

File: Kernel/Thread.h [hunk 2]
```diff
@@ -471,25 +471,26 @@ class Thread
...
+            TRY(taken_blockers.try_ensure_capacity(move_count));
             for (size_t i = 0; i < move_count; i++)
-                taken_blockers.append(m_blockers.take(i));
+                TRY(taken_blockers.try_append(m_blockers.take(i)));
             m_blockers.remove(0, move_count);
             return taken_blockers;
         }
 
-        void do_append_blockers(Vector<BlockerInfo, 4>&& blockers_to_append)
+        ErrorOr<void> do_append_blockers(Vector<BlockerInfo, 4>&& blockers_to_append)
```
Comment (bgianfo) [inline Kernel/Thread.h:484]: The fix here would be to use intrusive lists for all blockers so we don't have to use Vectors. As you can see we try to use Vectors with reserved capacity so you don't actually dynamically allocate, but it's finicky.

**Review by bgianfo — CHANGES_REQUESTED** [review]

File: Kernel/Locking/Mutex.cpp (hunk 1 above)
Comment (thomasqueirozb) [reply Kernel/Locking/Mutex.cpp:254]: It was my own `FIXME` from a previous commit lol. Should I just delete this commit then and revert to `MUST`ing those unblocks?

Comment (bgianfo) [reply Kernel/Locking/Mutex.cpp:254]: If you aren't interested in in fixing the allocations in unblock, then yea. Might be nice to add a FIXME to those functions so it's clear that we need to fix their implementations to not allocate.

**Comment by supercomputer7** [comment]
I am not fond of the idea that this PR just tries to fix a bunch of error propagation problems at once as it is much harder to review something like this. Still, the idea is much appreciated, but I rather push forward on making the kernel more stable
by taking this more lightly, because some problems need a bit more attention and just slapping a couple of `TRY()` just won't cut it.

For example, I opened a PR to fix the issue with `Device::after_inserting` method here:
https://github.com/SerenityOS/serenity/pull/16618
In that PR, I also try to find a solution I had for a bare metal problem I encountered. Some problems need an appropriate solution that sometimes arises only by experience and close look at the problem.
I will leave some comments but I think we should reconsider some parts of the PR, as well as to consider splitting this into multiple PRs.

File: Kernel/Memory/MemoryManager.cpp [hunk 3]
```diff
@@ -92,7 +93,9 @@ UNMAP_AFTER_INIT MemoryManager::MemoryManager()
 {
     s_the = this;
 
-    parse_memory_map();
+    if (auto parse_error = parse_memory_map(); parse_error.is_error())
+        PANIC("MM: failed to parse memory map: {}", parse_error.release_error());
```
Comment (supercomputer7) [inline Kernel/Memory/MemoryManager.cpp:97]: I am really not fond of this. If for some reason memory map parsing failed, I'd like it to fail on the exact line where it failed, which means we need to use `MUST()` - there's simple no reason to propagate an error back just to panic, which will make it much more harder to debug something in that part of the code.

**Comment by supercomputer7** [comment]
It's sad to see this being closed without a resolution of what to do further with this. I suggest splitting the commits for the sake of sending them again, but reviewing each one of them in a separate context.

**Pull request closed without merging**